site/docs/capabilities/llm-integrations/connect-providers.md

```diff
@@ -0,0 +1,507 @@
+    C -->|gpt-4o-mini| D[OpenAI AIServiceBackend]
+    C -->|claude-3-sonnet| E[Bedrock AIServiceBackend]
+    D --> F[OpenAI BackendSecurityPolicy]
+    E --> G[AWS BackendSecurityPolicy]
+    F --> H[OpenAI API]
+    G --> I[AWS Bedrock API]
+
+    style A fill:#e1f5fe
+    style B fill:#f3e5f5
+    style D fill:#e8f5e8
+    style E fill:#e8f5e8
+    style F fill:#fff3e0
+    style G fill:#fff3e0
+```
+
+### Data Flow Process
+
+1. **Request Reception**: Client sends a request to the AI Gateway with the OpenAI-compatible format
+2. **Route Matching**: AIGatewayRoute examines request headers (like `x-ai-eg-model`) to determine the target backend
+3. **Backend Resolution**: The matching rule identifies the appropriate AIServiceBackend
+4. **Authentication**: The AIServiceBackend's security policy provides credentials for upstream authentication
+5. **Schema Transformation**: If needed, the request is transformed from the input schema to the backend's expected schema
+6. **Provider Communication**: The request is forwarded to the actual AI provider with proper authentication
+7. **Response Processing**: The provider's response is transformed back to the unified schema format
+8. **Client Response**: The standardized response is returned to the client
+
+## Common Configuration Patterns
+
+### Single Provider Setup
+
+For a simple single-provider setup:
+
+```yaml
+# Backend configuration
+apiVersion: aigateway.envoyproxy.io/v1alpha1
+kind: AIServiceBackend
+metadata:
+  name: openai-backend
+spec:
+  schema:
+    name: OpenAI
+    version: v1
+  backendRef:
+    name: openai-backend
+    kind: Backend
+    group: gateway.envoyproxy.io
+  backendSecurityPolicyRef:
+    name: openai-auth
+    kind: BackendSecurityPolicy
+    group: aigateway.envoyproxy.io
+
+---
+# Security configuration
+apiVersion: aigateway.envoyproxy.io/v1alpha1
+kind: BackendSecurityPolicy
+metadata:
+  name: openai-auth
+spec:
+  type: APIKey
+  apiKey:
+    secretRef:
+      name: openai-secret
+      namespace: default
+
+---
+# Routing configuration
+apiVersion: aigateway.envoyproxy.io/v1alpha1
+kind: AIGatewayRoute
+metadata:
+  name: openai-route
+spec:
+  schema:
+    name: OpenAI
+    version: v1
+  parentRefs:
+    - name: my-gateway
+      kind: Gateway
+      group: gateway.networking.k8s.io
+  rules:
+    - backendRefs:
+        - name: openai-backend
+```
+
+### Multi-Provider Setup with Fallback
+
+For high availability with multiple providers:
+
+```yaml
+apiVersion: aigateway.envoyproxy.io/v1alpha1
+kind: AIGatewayRoute
+metadata:
+  name: multi-provider-fallback
+spec:
+  schema:
+    name: OpenAI
+    version: v1
+  parentRefs:
+    - name: my-gateway
+      kind: Gateway
+      group: gateway.networking.k8s.io
+  rules:
+    - matches:
+        - headers:
+            - type: Exact
+              name: x-ai-eg-model
+              value: gpt-4o-mini
+      backendRefs:
+        - name: openai-backend
+        - name: groq-backend  # Fallback provider
```

Comment:
i think maybe demonstrate the "modelNameOverride" API here https://deploy-preview-935--envoy-ai-gateway.netlify.app/docs/capabilities/traffic/model-name-virtualization/ since gpt-4o-mini won't exist with groq